File: src/java/magma/app/ConstructionCaller.java
package magma.app;

import magma.api.option.None;
import magma.api.option.Option;
import magma.app.compile.Caller;

record ConstructionCaller(String right, Platform platform) implements Caller {
    @Override
    public String generate() {
        if (Platform.Magma == this.platform) {
            return this.right;
        }
        return "new " + this.right;
    }

    @Override
    public Option<Value> findChild() {
        return new None<Value>();
    }
}

File: src/node/magma/app/ConstructionCaller.ts
import { Caller } from "../../magma/app/compile/Caller";
import { Platform } from "../../magma/app/Platform";
import { Value } from "../../magma/app/Value";
import { Option } from "../../magma/api/option/Option";
import { None } from "../../magma/api/option/None";
class ConstructionCaller implements Caller {
	right: string;
	platform: Platform;
	constructor (right: string, platform: Platform) {
		this.right = right;
		this.platform = platform;
	}
	generate(): string {
		if (Platform.Magma === this.platform) {
			return this.right;
		}
		return "new " + this.right;
	}
	findChild(): Option<Value> {
		return new None<Value>();
	}
}
